static/js/components/ShareDataForm.jsx

```diff
@@ -0,0 +1,224 @@
+    dispatch(spectraActions.fetchSourceSpectra(route.id));
+  }, [route.id, dispatch]);
+
+  const validateGroups = () => {
+    const formState = getValues({ nest: true });
+    return formState.groups.length >= 1;
+  };
+
+  const onSubmit = async (groupsFormData) => {
+    const selectedPhotIDs = selectedPhotRows.map((idx) => photometry[route.id][idx].id);
+    const selectedSpecIDs = selectedSpecRows.map((idx) => spectra[route.id][idx].id);
+    setIsSubmitting(true);
+    const data = {
+      groupIDs: groupsFormData.groups.map((g) => g.id),
+      photometryIDs: selectedPhotIDs,
+      spectrumIDs: selectedSpecIDs,
+    };
+    const result = await dispatch(sourceActions.shareData(data));
+    if (result.status === "success") {
+      dispatch(showNotification("Data successfully shared"));
+      reset({ groups: [] });
+      setSelectedPhotRows([]);
+      setSelectedSpecRows([]);
+    }
+    setIsSubmitting(false);
+  };
+
+  if (!photometry[route.id] || !allGroups || !spectra[route.id]) {
+    return <>Loading...</>;
+  }
+
+  const photRows = photometry[route.id].map((phot) => (
+    createPhotRow(phot.id, phot.mjd, phot.mag, phot.magerr, phot.limiting_mag,
+      phot.instrument_name, phot.filter)));
+
+  const specRows = spectra[route.id].map((spec) => (
+    createSpecRow(spec.id, spec.instrument_name)));
+
+  const options = {
+    textLabels: {
+      body: {
+        noMatch: "",
+      }
+    },
+    filter: true,
+    selectableRows: "multiple",
+    selectableRowsOnClick: true,
+    filterType: "dropdown",
+    responsive: "vertical",
+    rowsPerPage: 10,
+    selectableRowsHeader: true,
+    customToolbarSelect: () => { },
+    download: false,
+    print: false
+  };
+
+  return (
+    <>
+      <div>
+        <Typography variant="h5">
+          Share Source Data
+        </Typography>
+      </div>
+      <br />
+      <div>
+        <MUIDataTable
+          columns={photHeadCells}
+          data={photRows}
+          title="Photometry"
+          options={{
+            ...options,
+            rowsSelected: selectedPhotRows,
+            onRowSelectionChange: (rowsSelectedData, allRows, rowsSelected) => {
+              setSelectedPhotRows(rowsSelected);
+            }
+          }}
+        />
+        <br />
+        <MUIDataTable
+          columns={specHeadCells}
+          data={specRows}
+          title="Spectra"
+          options={{
+            ...options,
+            rowsSelected: selectedSpecRows,
+            onRowSelectionChange: (rowsSelectedData, allRows, rowsSelected) => {
+              setSelectedSpecRows(rowsSelected);
+            }
+          }}
+        />
+      </div>
+      <br />
+      <div>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          {
+            !!errors.groups &&
+              <FormValidationError message="Please select at least one group/user" />
+          }
+          <Controller
+            name="groups"
+            as={(
+              <Autocomplete
```

Comment:
Where are the single user groups (in load demo data)?

Reply:
Looks like they're not being set up properly in data_loader (Stéfan's PR fixes that)